### PR TITLE
Init repo

### PR DESCRIPTION
**Pull request by alshyra**
This PR contains:
* `broker-autobot` code
* Documentation
* Unit tests

Maybe we should add types later on

### CODE REVIEW

File: .prettierignore [hunk 1]
```diff
@@ -0,0 +1,4 @@
+helm
```
Comment (gautaz) [inline .prettierignore]: There won't be any helm file in this project

File: README.adoc [hunk 2]
```diff
@@ -107,8 +107,9 @@ To register an action to retrieve the common broker configuration:
 [source,javascript]
 ----
 const autobot = brokerAutobot({
-  retrieveAction: {
-    name: 'configuration-holder-service.get-broker-config',
+  settingsRetrieveAction: {
```
Comment (gautaz) [inline README.adoc]: Better name proposal: settingsRetrievalAction

File: package.json [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  "name": "broker-autobot",
+  "version": "0.0.0",
+  "description": "self-configuring robotic lifeform to manage centrally the configuration of all Moleculer brokers",
+  "main": "src",
+  "types": "./dist/typings.d.ts",
+  "directories": {
+    "test": "test"
+  },
+  "scripts": {
+    "test:debug": "ava debug --host 0.0.0.0 --port 8228",
```
Comment (gautaz) [inline package.json]: Only useful in a Docker context, to remove.

File: package.json [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    "test": "nyc ava",
+    "lint": "eslint ."
+  },
+  "nyc": {
```
Comment (gautaz) [inline package.json]: You can remove the nyc configuration, the default behavior should already work well.

File: package.json [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+  "bugs": {
+    "url": "https://github.com/r2d2bzh/broker-autobot/issues"
+  },
+  "homepage": "https://github.com/r2d2bzh/broker-autobot#readme",
+  "dependencies": {
+    "get-stream": "^6.0.0",
+    "lodash.merge": "^4.6.2",
+    "lodash.throttle": "^4.1.1",
+    "moleculer": "^0.14.11",
+    "nodemon": "^2.0.6",
```
Comment (gautaz) [inline package.json]: nodemon ? Not used.

File: package.json [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+    "uuid": "^8.3.1"
```
Comment (gautaz) [inline package.json]: uuid is a test dependency => devDependencies

File: src/index.js [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+// @ts-check
+const getStream = require('get-stream');
+const merge = require('lodash.merge');
+const newBrokerRevolver = require('./broker-revolver');
+const sleep = require('util').promisify(setTimeout);
+const identity = (x) => x;
+
+/**
+ * @param {brokerRevolver} brokerRevolver
+ * @param {settingsRetrieveAction} param1
```
Comment (gautaz) [inline src/index.js]: How about settingsRetrievalAction?

File: package.json [hunk 8]
```diff
@@ -0,0 +1,54 @@
+{
+  "name": "broker-autobot",
```
Comment (gautaz) [inline package.json]: @r2d2bzh/broker-autobot

**Review by gautaz — CHANGES_REQUESTED** [review]
A few details to change

File: .prettierignore (hunk 1 above)
Comment (alshyra) [reply .prettierignore]: the entire file actually

File: README.adoc (hunk 2 above)
Comment (alshyra) [reply README.adoc]: I replaced all occurences

File: package.json (hunk 3 above)
Comment (alshyra) [reply package.json]: Done

File: package.json (hunk 4 above)
Comment (alshyra) [reply package.json]: Done and it is still working fine

File: package.json (hunk 5 above)
Comment (alshyra) [reply package.json]: Removed

File: package.json (hunk 6 above)
Comment (alshyra) [reply package.json]: Moved

File: src/index.js (hunk 7 above)
Comment (alshyra) [reply src/index.js]: same as above

File: package.json (hunk 8 above)
Comment (alshyra) [reply package.json]: Renamed

**Review by gautaz — APPROVED** [review]
LGTM

**Comment by alshyra** [comment]
:+1:

**Pull request merged**